Mathlib/Tactic/Linarith/Parsing.lean

```diff
@@ -16,19 +15,43 @@ It identifies atoms up to ring-equivalence: that is, `(y*3)*x` will be identifie
 where the monomial `x*y` is the linear atom.
 
 * Variables are represented by natural numbers.
-* Monomials are represented by `Monom := RBMap ℕ ℕ`.
+* Monomials are represented by `Monom := TreeMap ℕ ℕ`.
   The monomial `1` is represented by the empty map.
-* Linear combinations of monomials are represented by `Sum := RBMap Monom ℤ`.
+* Linear combinations of monomials are represented by `Sum := TreeMap Monom ℤ`.
 
 All input expressions are converted to `Sum`s, preserving the map from expressions to variables.
 We then discard the monomial information, mapping each distinct monomial to a natural number.
-The resulting `RBMap ℕ ℤ` represents the ring-normalized linear form of the expression.
+The resulting `TreeMap ℕ ℤ` represents the ring-normalized linear form of the expression.
 This is ultimately converted into a `Linexp` in the obvious way.
 
 `linearFormsAndMaxVar` is the main entry point into this file. Everything else is contained.
 -/
 
 open Mathlib.Ineq Batteries
+open Std (TreeMap)
+
+namespace Std.TreeMap
+
+-- This will be replaced by a `BEq` instance implemented in the standard library,
```

Comment:
Can we leave a note somewhere to track this replacement? (I'd like to avoid a future where Std is changed and this is not removed. If you say "I'll keep it in mind", that could suffice!)